move/squid/sources/squid/coin_bag.move

```diff
@@ -41,6 +43,19 @@ module squid::coin_bag {
         }
     }
 
+    public(package) fun get_exact_balance<T>(self: &mut CoinBag, amount: u64): Option<Balance<T>> {
+        let key = get_balance_key<T>();
+
+        if(self.bag.contains(key)) {
+            let balance = self.bag.borrow_mut<address, Balance<T>>(key);
+            assert!(balance.value() >= amount, ENotEnoughBalance);
+
+            option::some(balance.split(amount))
+        } else {
+            option::none<Balance<T>>()
```

Comment:
probably revert here as well since 0 is not enough balance, and just return `Balance<T>` instead of `Option<Balance<T>>`.